### PR TITLE
feat: Feat/1673 bug report

### PR DESCRIPTION
**Pull request by Cal-L**
## **Description**

This PR adds a bug report section to the error boundary screen, which utilizes Sentry User Feedback API under the hood - https://docs.sentry.io/platforms/react-native/user-feedback/. The bug report will associate the issue ID, further assisting in debugging issues.

## **Related issues**

Fixes: [#1673](https://github.com/MetaMask/mobile-planning/issues/1673) , [#1448](https://github.com/MetaMask/mobile-planning/issues/1448)

## **Manual testing steps**

- Throw an error on the wallet screen
- Connect own Sentry instance
- Build app in release mode (enable auto source map upload)
- Open the app, navigate to the Wallet screen and notice the crash
- Report the bug
- See that a user feedback was reported in Sentry with the associated issue as reference

### **After**

Reporting the bug
https://github.com/MetaMask/metamask-mobile/assets/10508597/1179d801-3d61-4aa8-8fcd-3c743a12452a

User feedback in Sentry
https://github.com/MetaMask/metamask-mobile/assets/10508597/ede6bc57-0dda-450d-b833-60e6817564c7

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/Views/ErrorBoundary/index.js [hunk 1]
```diff
@@ -212,6 +250,8 @@ class ErrorBoundary extends Component {
   };
 
   componentDidCatch(error, errorInfo) {
+    const sentryId = getLatestSentryId();
```
Comment (Cal-L) [inline app/components/Views/ErrorBoundary/index.js:255]: Get's the ID of the latest captured exception

**Review by Daniel-Cross — DISMISSED** [review]
Looks good!

File: app/components/Views/ErrorBoundary/index.js [hunk 2]
```diff
@@ -108,6 +110,40 @@ const createStyles = (colors) =>
...
+        {'  '}
+        {strings('error_screen.submit_ticket_8')}{' '}
+        <Text onPress={promptBugReport} style={styles.link}>
+          {strings('error_screen.submit_ticket_6')}
+        </Text>{' '}
+        {strings('error_screen.submit_ticket_9')}
+      </Text>
+    ),
+    [promptBugReport, styles.text, styles.link, styles.reportStep],
+  );
```
Comment (jpuri) [inline app/components/Views/ErrorBoundary/index.js:146]: A callback to render jsx looks bit weird, I would rather prefer to put this piece inside jsx only.

**Review by NicolasMassart — COMMENTED** [review]
Q: What will we actually use this for given the current implementation only allows user to sends a feedback with the latest Sentry issue id, which I guess we already have logged in Sentry issues list, and doesn't allow to add a custom comment? Is this just a first step and we plan to add a custom input field later or add a screenshot?

File: app/components/Views/ErrorBoundary/index.js (hunk 2 above)
Comment (Cal-L) [reply app/components/Views/ErrorBoundary/index.js:142]: Updated!

**Review by tommasini — APPROVED** [review]
LGTM! Maybe this is better a isolated componen than a useCallback but will defer that decision to you

**Pull request merged**